crates/ruff_linter/src/rules/pyupgrade/rules/redundant_open_modes.rs

```diff
@@ -71,68 +70,71 @@ pub(crate) fn redundant_open_modes(checker: &mut Checker, call: &ast::ExprCall)
         return;
     }
 
-    let Some(mode_param) = call.arguments.find_argument_value("mode", 1) else {
+    let Some(mode_arg) = call.arguments.find_argument_value("mode", 1) else {
         return;
     };
-    let Expr::StringLiteral(ast::ExprStringLiteral { value, .. }) = &mode_param else {
+    let Expr::StringLiteral(ast::ExprStringLiteral { value, .. }) = &mode_arg else {
         return;
     };
     let Ok(mode) = OpenMode::from_chars(value.chars()) else {
         return;
     };
     let reduced = mode.reduce();
     if reduced != mode {
-        checker.diagnostics.push(create_diagnostic(
-            call,
-            mode_param,
-            reduced,
-            checker.tokens(),
-            checker.stylist(),
-        ));
+        checker
+            .diagnostics
+            .push(create_diagnostic(call, mode_arg, reduced, checker));
     }
 }
 
 fn create_diagnostic(
     call: &ast::ExprCall,
-    mode_param: &Expr,
+    mode_arg: &Expr,
     mode: OpenMode,
-    tokens: &Tokens,
-    stylist: &Stylist,
+    checker: &Checker,
 ) -> Diagnostic {
+    let range = if checker.settings.preview.is_enabled() {
+        mode_arg.range()
+    } else {
+        call.range
+    };
+
     let mut diagnostic = Diagnostic::new(
         RedundantOpenModes {
             replacement: mode.to_string(),
         },
-        call.range(),
+        range,
     );
 
     if mode.is_empty() {
-        diagnostic
-            .try_set_fix(|| create_remove_param_fix(call, mode_param, tokens).map(Fix::safe_edit));
+        diagnostic.try_set_fix(|| {
+            create_remove_argument_fix(call, mode_arg, checker.tokens()).map(Fix::safe_edit)
+        });
     } else {
+        let stylist = checker.stylist();
         diagnostic.set_fix(Fix::safe_edit(Edit::range_replacement(
             format!("{}{mode}{}", stylist.quote(), stylist.quote()),
-            mode_param.range(),
+            mode_arg.range(),
         )));
     }
 
     diagnostic
 }
 
-fn create_remove_param_fix(
+fn create_remove_argument_fix(
     call: &ast::ExprCall,
-    mode_param: &Expr,
+    mode_arg: &Expr,
     tokens: &Tokens,
 ) -> Result<Edit> {
-    // Find the last comma before mode_param and create a deletion fix
-    // starting from the comma and ending after mode_param.
+    // Find the last comma before mode_arg and create a deletion fix
```

Comment:
Maybe for another PR, should this use our generic `remove_argument` helper instead of rolling its own function for this?